File: array/1758_minimum_changes_to_make_alternating_binary_string/1758_minimum_changes_to_make_alternating_binary_string.py
class Solution:
    def minOperations(self, s: str) -> int:
        n=len(s)
        ans1=ans2=0
        for i in range(n):
            if i%2==0:
                if s[i]!='0':
                    ans2+=1
                else:
                    ans1+=1
            else:
                if s[i]!='0':
                    ans1+=1
                else:
                    ans2+=1
        return min(ans1,ans2)



s = "0100"
res = Solution().minOperations(s)
print(res)

File: array/1758_minimum_changes_to_make_alternating_binary_string/1758_minimum_changes_to_make_alternating_binary_string.go
package main

import "fmt"

func minOperations(s string) int {
	var ans1, ans2 int
	for i := range s {
		if i%2 == 0 {
			if string(s[i]) != "0" {
				ans2++
			} else {
				ans1++
			}
		} else {
			if string(s[i]) != "0" {
				ans1++
			} else {
				ans2++
			}
		}
	}
	return min(ans1, ans2)
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func main() {
	s := "0100"
	res := minOperations(s)
	fmt.Println(res)
}
